provisioner/file/provisioner.go

```diff
@@ -227,7 +232,7 @@ func (p *Provisioner) ProvisionUpload(ui packersdk.Ui, comm packersdk.Communicat
 
 		// If we're uploading a directory, short circuit and do that
 		if info.IsDir() {
-			if err = comm.UploadDir(dst, src, nil); err != nil {
+			if err = comm.UploadDir(dst, src, p.config.Excludes); err != nil {
```

Comment:
This change is needed in order to path in the list of files to exclude. But the actually implementation for supporting the excludes would need to be made in the Packer SDK communicator, which this line is calling out to. At this present time the exclude argument is ignored.  

https://github.com/hashicorp/packer-plugin-sdk/blob/b03e2b6b7bbc38ca6b92770b33873860e796a5e0/sdk-internals/communicator/ssh/communicator.go#L187

https://github.com/hashicorp/packer-plugin-sdk/blob/b03e2b6b7bbc38ca6b92770b33873860e796a5e0/sdk-internals/communicator/winrm/communicator.go#L149

For the excludes we will need to provide guidance on the allowed paths and/or globs that can be passed into exclude, which I think the file provisioner could take on the work of expanding.

Reply:
Thank you for noticing this. I already started a PR on packer-sdk to address this. I also left a comment there about how the pattern matching should be. Could you also take a look?

https://github.com/hashicorp/packer-plugin-sdk/pull/163